src/api/namespaces/atomicmarket/handlers/sales2.get-sales-templates.test.ts

```diff
@@ -0,0 +1,423 @@
+import 'mocha';
+import {expect} from 'chai';
+import {RequestValues} from '../../utils';
+import {initAtomicMarketTest} from '../test';
+import {getTestContext} from '../../../../utils/test';
+import {getSalesTemplatesV2Action} from './sales2';
+
+const {client, txit} = initAtomicMarketTest();
+
+async function getSalesIds(values: RequestValues): Promise<Array<number>> {
+    const testContext = getTestContext(client);
+
+    await client.query('SELECT update_atomicmarket_sales_filters()');
+
+    const result = await getSalesTemplatesV2Action({
+        symbol: 'TEST',
+        ...values,
+    }, testContext);
+
+    return result.map((s: any) => s.sale_id);
+}
+
+async function initTest(saleValues: Record<string, any> = {}): Promise<{template_id: number, sale_id: number}> {
+    const {template_id} = await client.createTemplate();
+
+    const {offer_id} = await client.createOfferAsset({}, {template_id});
+    const {sale_id} = await client.createSale({
+        offer_id,
+        ...saleValues,
+    });
+
+    return {
+        sale_id,
+        template_id: (await client.createTemplate()).template_id,
```

Comment:
Why are we creating two different templates here in this initialization?

Reply:
The sale created in initTest is (normally) not supposed to show in the results. Basically to make sure the filters are actually applied.
Since the query does a distinct on template_id, if it were using the same template_id, it would be possible that the filters are not applied, but the function still only returns the expected sale.
So this makes sure that if the filters are not applied, the test fails